### PR TITLE
Moving from CertificateSigningRequests to ServiceAccounts

### PR DESCRIPTION
**Pull request by filariow**
Depends on #126 and #130

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -11,8 +11,12 @@ rules:
   resources:
   - secrets
   verbs:
+  - create
+  - delete
   - get
   - list
+  - patch
```
Comment (Kartikey-star) [inline config/rbac/role.yaml:18]: Is Delete and Patch required for secrets ?

Comment (filariow) [reply config/rbac/role.yaml]: good catch! delete is required, patch can be removed. They are needed in order to allow the ClusterEnvironment controller to grant this permission to service agent's Service Account. However, we are not using Patch, so will remove it both from here and from Service Agent's requirements (i.e., `config/rbac/reporter_role.yaml`).

```
2023-04-26T12:30:00Z    ERROR   error binding namespace {"controller": "clusterenvironment", "controllerGroup": "primaza.io", "controllerKind": "ClusterEnvironment", "ClusterEnvironment": {"name":"worker","namespace":"primaza-system"}, "namespace": "primaza-system", "name": "worker", "reconcileID": "047815dc-0b32-4233-855b-c19629a839b0", "cluster-environment": "worker", "namespace": "services", "error": "rolebindings.rbac.authorization.k8s.io \"primaza-reporter-worker-services\" is forbidden: user \"system:serviceaccount:primaza-system:primaza-controller-manager\" (groups=[\"system:serviceaccounts\" \"system:serviceaccounts:primaza-system\" \"system:authenticated\"]) is attempting to grant RBAC permissions not currently held:\n{APIGroups:[\"\"], Resources:[\"secrets\"], Verbs:[\"create\" \"delete\" \"patch\" \"update\"]}"}
```

File: controllers/agents/svc/serviceclass_controller.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ import (
 // agents needs to write back registered services up to primaza.  It contains
 // two keys: `kubeconfig`, a serialized kubeconfig for the upstream kubeconfig
 // cluster, and `namespace`, the namespace to write registered services to
-const PRIMAZA_CONTROLLER_REFERENCE string = "primaza-kubeconfig"
```
Comment (Kartikey-star) [inline controllers/agents/svc/serviceclass_controller.go:49]: This constant has been moved to pkg/, so let's make this change there

Comment (filariow) [reply controllers/agents/svc/serviceclass_controller.go]: done

**Review by Kartikey-star — APPROVED** [review]
Looks good to me

**Pull request merged**